_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,366 @@
+
+```html
+<p>Helvetica is a widely used sans-serif typeface developed in 1957 by Max Miedinger and Eduard Hoffmann.</p>
+<p style="font-family: helvetica; background: #EEE; color: #777;" aria-hidden="true">
+	The quick brown fox jumps over the lazy dog.
+</p>
+```
+
+#### Passed Example 8
+
+This text has an insufficient contrast but it does not express anything in human language so it still passes.
+
+```html
+<p style="color: #000; background: #666;">
+	----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@
+</p>
+```
+
+#### Passed Example 9
+
+Even though the text is in a shadow DOM tree, the text is part of a child [text node](https://dom.spec.whatwg.org/#text) of an HTML in the flat tree and has sufficient contrast
+
+```html
+<p style="color: #333; background: #fff;" id="p"></p>
+<script>
+	const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+	shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This text has an insufficient contrast with the white background.
+
+```html
+<p style="color: #AAA; background: white;">
+	Some text in English
+</p>
+```
+
+#### Failed Example 2
+
+This text has an insufficient contrast with the darkest point on the gradient background behind the text.
+
+```html
+<p style="color: #AAA; background: linear-gradient(to right, #FFF, #00F); width: 300px">
+	Some text in English
+</p>
+```
+
+#### Failed Example 3
+
+This text has an insufficient contrast with the background image.
+
+```html
+<p
+	style="color: #555; height:50px; padding-top:25px; background: black no-repeat -20px -20px url('../test-assets/contrast/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Failed Example 4
+
+This text fails because the alpha transparency significantly lowers the contrast of the otherwise black text.
+
+```html
+<p style="color: rgba(0,0,0,.3); background: #FFF">
+	Some text in English
+</p>
+```
+
+#### Failed Example 5
+
+This text fails because the CSS opacity property significantly lowers the contrast of the otherwise black text.
+
+```html
+<div style="background: #FFF">
+	<p style="color: #000; opacity: .3">
+		Some text in English
+	</p>
+</div>
+```
+
+#### Failed Example 6
+
+Even though the text is in a shadow DOM tree, the text is part of a child [text node](https://dom.spec.whatwg.org/#text) of an HTML in the flat tree and has insufficient contrast
+
+```html
+<p style="color: #aaa; background: #fff;" id="p"></p>
+<script>
+	const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+	shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+#### Failed Example 7
+
+The text is placed over a black and white background, The grey text fails because of its opacity, where the text on the black background becomes too dark, and the text on the white background becomes too light.
```

Comment:
```suggestion
The text is placed over a black and white background. The grey text fails because of its opacity, where the text on the black background becomes too dark, and the text on the white background becomes too light.
```